src/meta/api/src/schema_api_impl.rs

```diff
@@ -2301,15 +2297,9 @@ impl<KV: kvapi::KVApi<Error = MetaError> + ?Sized> SchemaApi for KV {
         }
 
         let seq_metas = self.mget_kv(&meta_kv_keys).await?;
-        if seq_metas.len() != table_names.len() {
```

Comment:
drop table in the databend doesn't actually drop the table, it just changes drop_on to not None in tablemeta

In grant kv, the behavior of drop table cannot be perceived. This requires filtering on show grants, resulting in the function mget_table_name_by_ids.

In func  mget_table_name_by_ids

```
table_names is generate by mget DBIdTableName. the id is function's arg table_id.

seq_metas is generated by mget TableId.
```

I assume that seq_name and table_names must be the same length because the table ids are the same. But in fact, after the added judgment, resulting in an error.

Hi @drmingdrmer xp, is there anything that might have made a difference here?

Reply:
what does the flatten mean at the line 2288?

as far as I know, meet always return the exactly the same number of rows of the input keys. 
I’m not quite sure what happened. Is that possible to print out the two vector so we can see what the different is

Reply:
One word wakes up a dreamer. Instead of ignoring None, you should fill in none. Otherwise, creating an artificial inconsistency.